arcade/camera/camera_2d.py

```diff
@@ -0,0 +1,713 @@
+
+    @property
+    def projection_bottom_scaled(self) -> float:
+        """
+        The bottom edge of the projection in world space.
+        This is not adjusted with the camera position.
+
+        NOTE this IS scaled by zoom.
+        If this isn't what you want,
+        use projection_bottom instead.
+        """
+        return self._projection.bottom / self._data.zoom
+
+    @projection_bottom_scaled.setter
+    def projection_bottom_scaled(self, _bottom: float) -> None:
+        self._projection.bottom = _bottom * self._data.zoom
+
+    @property
+    def projection_top(self) -> float:
+        """
+        The top edge of the projection in world space.
+        This is not adjusted with the camera position.
+
+        NOTE this IS NOT scaled by zoom.
+        If this isn't what you want,
+        use projection_top_scaled instead.
+        """
+        return self._projection.top
+
+    @projection_top.setter
+    def projection_top(self, _top: float) -> None:
+        self._projection.top = _top
+
+    @property
+    def projection_top_scaled(self) -> float:
+        """
+        The top edge of the projection in world space.
+        This is not adjusted with the camera position.
+
+        NOTE this IS scaled by zoom.
+        If this isn't what you want,
+        use projection_top instead.
+        """
+        return self._projection.top / self._data.zoom
+
+    @projection_top_scaled.setter
+    def projection_top_scaled(self, _top: float) -> None:
+        self._projection.top = _top * self._data.zoom
+
+    @property
+    def projection_near(self) -> float:
+        """
+        The near plane of the projection in world space.
+        This is not adjusted with the camera position.
+
+        NOTE this IS NOT scaled by zoom.
+        """
+        return self._projection.near
+
+    @projection_near.setter
+    def projection_near(self, _near: float) -> None:
+        self._projection.near = _near
+
+    @property
+    def projection_far(self) -> float:
+        """
+        The far plane of the projection in world space.
+        This is not adjusted with the camera position.
+
+        NOTE this IS NOT scaled by zoom.
+        """
+        return self._projection.far
+
+    @projection_far.setter
+    def projection_far(self, _far: float) -> None:
+        self._projection.far = _far
+
+    @property
+    def viewport(self) -> Tuple[int, int, int, int]:
+        """
+        The pixel area that will be drawn to while the camera is active.
+        (left, right, bottom, top)
+        """
+        return self._projection.viewport
+
+    @viewport.setter
+    def viewport(self, _viewport: Tuple[int, int, int, int]) -> None:
+        self._projection.viewport = _viewport
+
+    @property
+    def viewport_width(self) -> int:
+        """
+        The width of the viewport.
+        Defines the number of pixels drawn too horizontally.
+        """
+        return self._projection.viewport[2]
```

Comment:
I think this is the viewport width only when the viewport starts at (0,0) (e.g. (0, 0, x, y), but this will produce an incorrect width when the viewport starts at anything other than the origin. Should it be something like:

`return self._projection.viewport[2] - self._projection.viewport[0]`

Reply:
See the previous comment with the viewport being stored as left, bottom, width, and height.